### PR TITLE
Expirity date functionality - The products should stop listing if the expiry date has passed

### PR DESCRIPTION
**Pull request by derrick-nuby**
# Product Expiry Functionality

## What does this PR do?
This PR implements the functionality for handling product expiry dates. When a product's expiration date has passed, the product is marked as expired and removed from the list of available products.

## Description of Task Completed
The task involves updating product records based on their expiration date:
- When the expiration date has passed, the product's availability flag (`isAvailable`) is turned off.
- The product is removed from the list of available products.
- The product owner is notified by email

## How should this be manually tested?
This functionality can be tested manually by:
1. Creating a product with an expiration date in the past.
2. Checking that the product's availability status is updated correctly.
3. Verifying that the expired product is no longer listed in the available products.

## Any background context you want to provide?
The implementation uses Node.js and Sequelize ORM to interact with the database. It leverages cron jobs to check and process expired products automatically after 24hrs.

### Setting up the Cron Job
- A cron job (`expiredProductsCron.js`) is scheduled to run after 24 hrs and process expired products
- i runs when the time hits 00:00
- the cronjobs searches for products that have expiryDate less than the current date(`expiredProducts `) then updates their isAvailable status to false.
- code example below
- sending email and flaggin as false can be found in the code

```javascript
cron.schedule('0 0 * * *', async () => {
    const currentDateOnly = new Date().toISOString().slice(0, 10);
    try {
        const expiredProducts = await Product.findAll({
            where: {
                expiryDate: {
                    [Op.lt]: Sequelize.literal(`DATE('${currentDateOnly}')`),
                },
                isAvailable: true,
            },
        });
```

### CODE REVIEW

File: src/cronjobs/expiredProductsCron.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        const expiredProducts = await Product.findAll({
+            where: {
+                expiryDate: {
+                    [Op.lt]: Sequelize.literal(`DATE('${currentDateOnly}')`),
+                },
+                isAvailable: true,
+            },
+        });
+
+        console.log(`Found ${expiredProducts.length} expired products.`);
```
Comment (kirengamartial) [inline src/cronjobs/expiredProductsCron.ts]: you may remove this console.log() is not necessary in production

File: src/cronjobs/expiredProductsCron.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+        await Promise.all(expiredProducts.map(async (product) => {
+            try {
+
+                console.log(`The product being processed is`, product.dataValues);
```
Comment (kirengamartial) [inline src/cronjobs/expiredProductsCron.ts]: even here

**Review by kirengamartial — CHANGES_REQUESTED** [review]
resolve those conflicts

and look for this feedback and address it

**Review by Balinda21 — CHANGES_REQUESTED** [review]
solve the conflicts, 
otherwise the codes looks good

**Review by kirengamartial — APPROVED** [review]

**Review by Balinda21 — APPROVED** [review]
Well done

**Review by Ishimwe7 — APPROVED** [review]
I like the approach you used for this task, keep it up .

**Review by bahati10 — APPROVED** [review]
Fix the conflicts otherwise everything looks good and ready

**Review by robsdagreat — APPROVED** [review]
Everything looks good except for the conflicts. Do resolve them please

**Pull request merged**